misc/python/materialize/bintray.py

```diff
@@ -0,0 +1,146 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""A Bintray API client.
+
+Consult the [official Bintray API documentation][official-docs] for details.
+Only the operations that are presently necessary are implemented.
+
+[official-docs]: https://bintray.com/docs/api/
+"""
+
+from typing import IO, Sequence
+from typing_extensions import Final
+from requests import Session, Response
+import requests
+
+API_BASE: Final = "https://api.bintray.com"
+
+HTTPError = requests.HTTPError
+
+
+class VersionAlreadyExistsError(Exception):
+    """Raised when creating a version of a Bintray package that already
+    exists."""
+
+    pass
+
+
+class PackageClient:
+    """A client for a specific Bintray package."""
+
+    def __init__(self, session: Session, subject: str, repo: str, package: str):
```

Comment:
this should `-> None` to get mypy to typecheck it. 😢

Reply:
I think that if `deny_untyped_defs` is turned on in the mypy configuration then you'll get errors for things that aren't checked.

Reply:
`deny_untyped_defs` explodes horribly on decoding JSON and YAML unfortunately. Haven't figured out a good way around that yet. Do you have suggestions? There doesn't seem to be much prior art on typed JSON/YAML decoding in Python.